### PR TITLE
dnf: Avoid downloading caches multiple times in one call

### PR DESCRIPTION
**Pull request by aleasto**

### CODE REVIEW

**Comment by aleasto** [comment]
The issue here is that `dnf_utils_create_sack_for_filters` will eventually try to update the caches again.

To be honest, I'm not convinced that honouring the `cache-age` hint in `RefreshCache` makes any sense: we are explicitly asking to refresh the cache... But of the two this change is the less intrusive one.

@Conan-Kudo @gordonmessmer @hughsie

**Review by Conan-Kudo — APPROVED** [review]
I think this is probably reasonable.

File: backends/dnf/pk-backend-dnf.c [hunk 1]
```diff
@@ -1706,6 +1706,9 @@ pk_backend_refresh_cache_thread (PkBackendJob *job,
 	/* invalidate the sack cache after downloading new metadata */
 	pk_backend_sack_cache_invalidate (backend, "downloaded new metadata");
 
+	// We just downloaded our cache, avoid doing so again
```
Comment (ximion) [inline backends/dnf/pk-backend-dnf.c:1709]:
```suggestion
	/* We just downloaded our cache, avoid doing so again */
```

**Review by ximion — COMMENTED** [review]
Just a style nitpick, otherwise I think this is sensible...

**Pull request merged**